components/date_picker/date_picker.tsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+
+import FocusTrap from 'focus-trap-react';
```

Comment:
We are avoiding adding new libraries for now, unless its a unicorn library that does exactly what we need, and we are even getting rid of a lot currently. So let's try to use a custom solution here if possible. Sorry for not catching this earlier!

Reply:
We are avoiding adding new libraries for now, unless its a unicorn library that does exactly what we need, and we are even getting rid of a lot currently. So let's try to use a custom solution here if possible. Sorry for not catching this earlier!

Reply:
I'm surprised that's not part of the popper library itself. I'm pretty sure even our old version of React Bootstrap handles that itself.

It looks like Floating UI has support for this via its `FloatingFocusManager`, so maybe we should switch this to use Floating UI instead of React Popper?

Reply:
Actually, yeah, let's use Floating UI here. We actually only have 1 other place where we use react-popper, so let's avoid spreading that to keep the number of those libraries down